Networking/Networking/Mapper/ShipmentTrackingProviderListMapper.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+/// (Attempts) to convert a dictionary into an ShipmentTrackingProviderGroup entity.
+///
+struct ShipmentTrackingProviderListMapper: Mapper {
+    func map(response: Data) throws -> [ShipmentTrackingProviderGroup] {
+        let decoder = JSONDecoder()
+        let rawDictionary = try? decoder.decode(ShipmentTrackingProviderListEnvelope.self, from: response).rawData
+        return rawDictionary?.map({ ShipmentTrackingProviderGroup(name: $0.key, dictionary: $0.value) }) ?? []
+    }
+}
+
+
+/// ShipmentTrackingProviderListEnvelope Disposable Entity: The shipment tracking provider endpoint returns
+/// the providers within a `data` key.
+///
+private struct ShipmentTrackingProviderListEnvelope: Decodable {
+    let rawData: [String: [String: String]]
```

Comment:
I think this should be
```let group: [ShipmentTrackingGroup]``` ?

Reply:
@mindgraffiti in a perfect world, yes. Unfortunately because of the way the JSON data for tracking providers is structured, `ShipmentTrackingGroup` does not conform to `Codable`. If you look at an example JSON response:

```json
{
  "data": {
    "Australia": {
      "Australia Post": "http:\/\/auspost.com.au\/track\/track.html?id=%1$s",
      "Fastway Couriers": "http:\/\/www.fastway.com.au\/courier-services\/track-your-parcel?l=%1$s"
    },
    "Austria": {
      "post.at": "http:\/\/www.post.at\/sendungsverfolgung.php?pnum1=%1$s",
      "dhl.at": "http:\/\/www.dhl.at\/content\/at\/de\/express\/sendungsverfolgung.html?brand=DHL&AWB=%1$s",
      "DPD.at": "https:\/\/tracking.dpd.de\/parcelstatus?locale=de_AT&query=%1$s"
    },
    ...
  }
}
```

The only key which is static is `data` — all the other keys are unknown until the data is fetched. Thus defining a set of `CodingKeys` inside of `ShipmentTrackingGroup` would be difficult. So the workaround to all of this is to simply define a dictionary of dictionaries (`[String: [String: String]]`) and parsing it the best we can. 

`¯\_(ツ)_/¯`

Reply:
Oh booooo. I see what you mean. It _has_ to be hard-coded like that because the fields are unknowns. Thanks for the clarification.